sdk/core/Azure.Core.Experimental/src/CloudMachine.cs

```diff
@@ -0,0 +1,253 @@
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new InvalidCloudMachineConfigurationException(configurationFile, setting: null, e);
+            }
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from stream.
+        /// </summary>
+        /// <param name="configurationContent"></param>
+        /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
+        public CloudMachine(Stream configurationContent)
+        {
+            try
+            {
+                var document = JsonDocument.Parse(configurationContent);
+                JsonElement json = document.RootElement.GetProperty("CloudMachine");
+
+                Id = ReaderString(json, "id", nameof(configurationContent));
+                SubscriptionId = ReaderString(json, "subscriptionId", nameof(configurationContent));
+                Region = ReaderString(json, "region", nameof(configurationContent));
+                DisplayName = ReaderString(json, "name", nameof(configurationContent));
+                Version = ReaderInt32(json, "version", nameof(configurationContent));
+            }
+            catch (InvalidCloudMachineConfigurationException)
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new InvalidCloudMachineConfigurationException(nameof(configurationContent), setting: null, e);
+            }
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from configuration system
+        /// </summary>
+        public CloudMachine(IConfiguration configuration)
+        {
+            try {
+                Id = ReadString(configuration, "CloudMachine:id");
+                SubscriptionId = ReadString(configuration, "CloudMachine:subscriptionId");
+                Region = ReadString(configuration, "CloudMachine:region");
+                DisplayName = ReadString(configuration, "CloudMachine:name");
+                Version = ReadInt32(configuration, "CloudMachine:version");
+            }
+            catch (InvalidCloudMachineConfigurationException) {
+                throw;
+            }
+            catch (Exception e) {
+                throw new InvalidCloudMachineConfigurationException(nameof(IConfiguration), setting: null, e);
+            }
+        }
+
+        private CloudMachine(string id, string displayName, string subscriptionId, string region, int version)
+        {
+            if (string.IsNullOrEmpty(id)) throw new ArgumentNullException(nameof(id));
+            if (string.IsNullOrEmpty(displayName)) throw new ArgumentNullException(nameof(displayName));
+            if (string.IsNullOrEmpty(subscriptionId)) throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrEmpty(region)) throw new ArgumentNullException(nameof(region));
+            if (version<0) throw new ArgumentOutOfRangeException(nameof(version));
```

Comment:
nit: format doc for spacing

Reply:
sorry, which line?

Reply:
```c#
if (version < 0) throw new ArgumentOutOfRangeException(nameof(version));
```

Reply:
Ah, makes sense. I did not understand what "doc" referred to.